tests/unit/test_coresubset.py

```diff
@@ -116,6 +116,7 @@ def test_fit_comparison_to_random_and_refined(self) -> None:
         # to typically compete with a kernel herding approach
         generator = np.random.default_rng(self.random_data_generation_key)
         x = generator.random((num_data_points, self.dimension))
+        metric_x = coreax.data.Data(x)
```

Comment:
Bit hacky but can be cleaned up when reduction is changed to use the new data classes

Reply:
Should add a ticket for the reduction refactoring, and mention on it that this change needs to made

Reply:
Added basic ticket for this #612